QTeachersApp/View/loginwindow.cpp

```diff
@@ -0,0 +1,58 @@
+#include "loginwindow.h"
+#include "ui_loginwindow.h"
+
+namespace HubertiusNamespace
+{
+    LoginWindow::LoginWindow(QWidget *parent):
+        QMainWindow(parent),
+        ui(new Ui::LoginWindow)
+    {
+        ui->setupUi(this);
+        myDatabase = QSqlDatabase::addDatabase("QSQLITE");
+        QDir tempPath = QDir::current();
+        tempPath.cdUp();  // to get out from /.../build-QTeachersApp-Desktop-Debug one folder up into /All-Teachers-In-System
+        qDebug() << tempPath.path() + "/QTeachersApp/DataBases/login.sqlite";
+        connOpen(&myDatabase, tempPath.path() + "/QTeachersApp/DataBases/login.sqlite");
+    }
+
+    LoginWindow::~LoginWindow()
+    {
+        delete ui;
+    }
+
+
+    void LoginWindow::on_btnLogin_clicked()
+    {
+
+        QString Username, Password;
+        Username = ui->lineEditUsername->text();
+        Password = ui->lineEditPassword->text();
+
+        if(myDatabase.isOpen())
+        {
+            QSqlQuery query("SELECT * FROM Login WHERE Username = '"+Username+"' AND Password = '"+Password+"'");
```

Comment:
Wprowadźmy dwie nowe rzeczy:

1. Zapobieganie SQL Injection. W skrócie, jeśli nie robimy żadnego escape/query preparation, to umożliwiamy użytkownikowi wykonanie niecnych operacji poprzez przekazywane przez niego wartości - takie jak np. dropnięcie całej tabeli. 
Dobrze to obrazuje ten komiks:
https://xkcd.com/327/  
Zadanie to sprawić, żeby wartości od użytkownika były bezpieczne. :wink: 
2. Hashowanie haseł. Generalnie, musimy zawsze starać się, by dane użytkownika były bezpieczne. Ale czasem i tak zdarzy się wyciek. Hasła nie powinny być widoczne w bazie danych jako plaintext, tzn. nikt nie powinien być w stanie ich odczytać i od razu wiedzieć co ma wpisywać. :slightly_smiling_face: 
Poszukaj informacji o "password encryption" i hashuj hasła :smile: 


Plus jeszcze hasła nie powinny być widoczne w repo - powywalaj je. Tak samo wszelkie tokeny, sekrety etc. (to ostatnie już bardziej na przyszłość)